Models.js

```diff
@@ -0,0 +1,136 @@
+function getExp(skillName){
+  /* Gets exp of a skill from storage.
+   */
+  return new Promise((resolve, reject) => {
+    chrome.storage.sync.get([skillName], function (result) {
+      var overallExp;
+      if (skillName in result) {
+        overallExp = result[skillName];
+      }
+      else{
+        overallExp = 0;
+      }
+      resolve(overallExp);
+    });
+  });
+}
+
+function getLevelAndLevelUpExp(exp) {
+  /* Calculates current level and exp needed to next level.
+   * O(max_level) - computational complexity
+   */
+  var level = 0;
+  while(exp >= expTable[level + 1]){
+    level++;
+  }
+  var levelExp = exp - expTable[level];
+  var totalExpNeeded = expTable[level + 1] - expTable[level];
+  var levelUpExp = totalExpNeeded - levelExp;
+  var returnObject = [level, levelUpExp];
+  return returnObject;
+}
+
+function addSkill () {
+  /* Adds skill to storage and to current table.
+   */
+  var skillName = $('#skill_name').val();
+  $('#skill_name').val('');
+
+  var skillsDict = {};
+  skillsDict[skillName] = 0;
+  allSkills.push(skillName);
+
+  chrome.storage.sync.set({skillsArrayId: allSkills});
+  chrome.storage.sync.set(skillsDict);
+  debugAddingSkill(skillName);
+  skillToTable(allSkills.length - 1);
+}
+
+function getSkills () {
+  /*  Creates a local table by taking skills from chrome storage.
+   */
+  return new Promise((resolve, reject) => {
+    var setSkillsArray = function (result) {
+      if(skillsArrayId in result) {
+        allSkills = result[skillsArrayId];
+        resolve();
+      }
+      else{
+        extension_log("There isn't any array of skills` names.");
+        reject();
+      }
+    };
+    chrome.storage.sync.get([skillsArrayId], setSkillsArray);
+  });
+}
+
+function clearSkills () {
+  /* Erases all skills.
+   */
+  var emptyArray = [];
+  chrome.storage.sync.set({skillsArrayId: emptyArray});
+}
+
+function updateSkill(skillNr){
+  /* Increases skill's exp by the amount in correspondent text area.
+   */
+  return new Promise((resolve, reject) => {
+    var addedExp = parseInt($("#add_value_num" + skillNr).val());
+    $("add_value_num" + skillNr).val('');
+    var skillName = allSkills[skillNr];
+
+    var updateStorage = function (result) {
+      var overallExp = 0;
+      if (skillName in result) {
+        overallExp += result[skillName] + addedExp;
+      }
+      else {
+        overallExp += addedExp;
+      }
+      var skillsDict = {};
+      skillsDict[skillName] = overallExp;
+      chrome.storage.sync.set(skillsDict, function(){
+        resolve(skillNr);
+      });
+    };
+    chrome.storage.sync.get([skillName], updateStorage);
+  });
+}
+
+function handleSkillButtons () {
+  /* Manages event listeners corresponding to skills.
+ 	 */
+  $("#skills").on("click", ".add_value_buttons", function () {
+		var skillNr = this.id.replace('add_value_button', '');
```

Comment:
Fail indentation.

Reply:
Strange... My indentation isn't failed.
![screenshot from 2017-12-17 20-28-49](https://user-images.githubusercontent.com/13454526/34083030-54392710-e369-11e7-9fd8-021e7138aa16.png)

Reply:
There are tabs in the code instead of spaces. To check it you can type in bash `grep -P '\t' Models.js`.